### PR TITLE
tgui: Interface improvements

### PR DESCRIPTION
**Pull request by Arkatos1**
## About The Pull Request

This PR brings various small improvements and cleanup to my previous interfaces, bundling them together to limit conflicts with other PRs.

**Stack Menu Interface**

1. Added a search bar.
2. Better spacing.
3. Small backend cleanup.

StackSearch

**Teleporter Interface**

1. Improved and more compact layout.

NewTeleporter

**Shuttle Console Interface**

1. Small margin tweaks, so window does not shift upon dropdown selection.
2. Cleaned up backend, which reduces number of ui_acts on subtypes and copypasta.
3. UI will now show recharging shuttle status.

**Exosuit Control Console Interface**

1. Shortened "Send Message" to just "Message".
2. Now supports showing multiple pilots as per newest mecha refactor.

**Holopad Interface** - Small grammar fix.

**Turret Control Inteface** - Window height now scales depending if you are a silicon lifeform or not.

**Particle Accelerator Interface** - Removed, as the object itself was removed some time ago. Sleep tight, sweet prince of singularities.

## Why It's Good For The Game

Better UIs.

## Changelog
:cl: Arkatos
add: Added a search bar to the Stack Menu UI.
tweak: Improvements to the Teleporter UI layout.
tweak: Turret Control UI window height will now scale depending if you are a silicon or not.
tweak: Shuttle Console UI will now properly show recharging status, and will not wildly shift window upon dropdown selection.
fix: Exosuit Control Console UI will now properly show multiple pilots.
spellcheck: Fixed a minor grammar mistake in the Holopad UI.
/:cl:

### CODE REVIEW

**Comment by Jared-Fogle** [comment]
Love the stack search autofocusing :+1:

File: tgui/packages/tgui/interfaces/ShuttleConsole.js [hunk 1]
```diff
@@ -96,7 +96,9 @@ const ShuttleConsoleContent = (props, context) => {
               ? 'average'
               : status==="Igniting"
                 ? 'average'
-                : 'bad'}
+                : status==="Recharging"
+                  ? 'average'
+                  : 'bad'}
```
Comment (Jared-Fogle) [inline tgui/packages/tgui/interfaces/ShuttleConsole.js:101]: This is a nasty ternary. Can you split this out into a switch/case?

Comment (Arkatos1) [reply tgui/packages/tgui/interfaces/ShuttleConsole.js]: Commited a much cleaner solution with a color keys instead, which allows to easily modify existing colors for the each status, or add a new one.

Comment (Jared-Fogle) [reply tgui/packages/tgui/interfaces/ShuttleConsole.js]: Works for me!

**Pull request merged**